components/Editor/ExtraColumn.tsx

```diff
@@ -60,7 +65,17 @@ const ExtraColumn = ({
             instructions={sierraStatements}
             codeType={codeType}
             activeIndexes={
-              casmToSierraProgramMap[activeCasmInstructionIndex] ?? []
+              casmToSierraProgramMap[activeCasmInstructionIndex]
+                ? debugMode === ProgramDebugMode.Sierra
+                  ? sierraSubStepIndex !== undefined
+                    ? [
+                        casmToSierraProgramMap[activeCasmInstructionIndex][
+                          sierraSubStepIndex
+                        ],
+                      ]
+                    : []
+                  : casmToSierraProgramMap[activeCasmInstructionIndex]
+                : []
```

Comment:
Please move this out of the layout and set it to a variable like `activeSierraIndexes`. Add comments as well. Do this in other places too to make the code more readable.

Reply:
Gotcha, should i add it in cairoVMApiContext?

Reply:
Pushed!